Info-Orbs/include/widgets/stockWidget.h

```diff
@@ -1,38 +1,32 @@
 #ifndef STOCK_WIDGET_H
 #define STOCK_WIDGET_H
 
-#include "widget.h"
-#include <vector>
 #include <ArduinoJson.h>
 #include <HTTPClient.h>
 #include <TFT_eSPI.h>
 
+#include "stockDataModel.h"
+#include "widget.h"
+
+#define MAX_STOCKS 5
+
 class StockWidget : public Widget {
-public:
+   public:
     StockWidget(ScreenManager &manager);
     void setup() override;
     void update(bool force = false) override;
     void draw(bool force = false) override;
     void changeMode() override;
 
-private:
-    void getStockData();
-    void displayStock(int displayIndex, uint32_t backgroundColor, uint32_t textColor);
+   private:
+    void getStockData(StockDataModel &stock);
+    void displayStock(int8_t displayIndex, StockDataModel &stock, uint32_t backgroundColor, uint32_t textColor);
 
-    unsigned long m_stockDelay;
-    unsigned long m_stockDelayPrev;
-    int currentStockIndex;
-    std::vector<String> stockTickers;
+    String m_apiKey = "c1tjb52ad3ia4h4uee9g";
```

Comment:
We should be careful about putting API keys in git - I've done this before and have royally horked the service it was for (spoiler: it was discord. The server owners were very unhappy with me)

Reply:
Fair concern. I am just moving it here from being hardcoded into the url. This is Brett's call as he intends on paying for the API access and when I suggested that the weather api key should be a user configured value he wanted it hardcoded. So I split the difference here by making it easy to change but still hardcoded.

Reply:
But I did go ahead and move it into the config.h.template like the weather api key. That way it is at least not in the code directly. Maybe we should talk Brett into distributing it outside of git.